srml/aura/src/lib.rs

```diff
@@ -48,15 +47,90 @@ extern crate lazy_static;
 #[cfg(test)]
 extern crate parking_lot;
 
-use rstd::prelude::*;
+use rstd::{result, prelude::*};
 use runtime_support::storage::StorageValue;
-use runtime_support::dispatch::Result;
 use primitives::traits::{As, Zero};
 use timestamp::OnTimestampSet;
+#[cfg(feature = "std")]
+use timestamp::TimestampInherentData;
+#[cfg(feature = "std")]
+use parity_codec::Decode;
+use inherents::{RuntimeString, InherentIdentifier, InherentData, ProvideInherent, MakeFatalError};
+#[cfg(feature = "std")]
+use inherents::{InherentDataProviders, ProvideInherentData};
 
 mod mock;
 mod tests;
 
+/// The aura inherent identifier.
+pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"auraslot";
+
+/// The type of the aura inherent.
+pub type InherentType = u64;
+
+/// Auxiliary trait to extract aura inherent data.
+pub trait AuraInherentData {
+	/// Get aura inherent data.
+	fn aura_inherent_data(&self) -> result::Result<InherentType, RuntimeString>;
+	/// Replace aura inherent data.
+	fn aura_replace_inherent_data(&mut self, new: InherentType);
+}
+
+impl AuraInherentData for InherentData {
+	fn aura_inherent_data(&self) -> result::Result<InherentType, RuntimeString> {
+		self.get_data(&INHERENT_IDENTIFIER)
+			.and_then(|r| r.ok_or_else(|| "Aura inherent data not found".into()))
+	}
+
+	fn aura_replace_inherent_data(&mut self, new: InherentType) {
+		self.replace_data(INHERENT_IDENTIFIER, &new);
+	}
+}
+
+#[cfg(feature = "std")]
+pub struct InherentDataProvider {
+	slot_duration: u64,
+}
+
+#[cfg(feature = "std")]
+impl InherentDataProvider {
+	pub fn new(slot_duration: u64) -> Self {
+		Self {
+			slot_duration
+		}
+	}
+}
+
+#[cfg(feature = "std")]
+impl ProvideInherentData for InherentDataProvider {
+	fn on_register(
+		&self, providers: &InherentDataProviders
```

Comment:
formatting on this code and some others in this `impl` block is wrong; all-or-nothing args indent.